### PR TITLE
chore: upgrade dfnity libraries to v10

### PR DESCRIPTION
**Pull request by 0xAaCE**

### CODE REVIEW

File: src/utils/buffer.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  const result = new Uint8Array(
+    buffers.reduce((acc, curr) => acc + curr.byteLength, 0)
+  );
+  let index = 0;
+  for (const b of buffers) {
+    result.set(new Uint8Array(b), index);
+    index += b.byteLength;
+  }
+  return result.buffer;
+}
```
Comment (rocky-fleek) [inline src/utils/buffer.ts:15]: image

**Review by rocky-fleek — APPROVED** [review]
Is it ok to use ArrayBuffers like that? Why taht change?

**Review by punkbit — APPROVED** [review]
@aleacevedo had a look, noticed the move to dfinity 0.0.10, and confirm the changes for the move of types from dfinity/candid (Blob to ArrayBuffer, etc), but haven't tested locally with a plug build.

**Comment by rocky-fleek** [comment]
Oh OK so dfinity v0.0.10 finally moves to ArrayBuffers and removes all the blobToUint8Array and stuff. Nice

**Comment by 0xAaCE** [comment]
> Oh OK so dfinity v0.0.10 finally moves to ArrayBuffers and removes all the blobToUint8Array and stuff. Nice

Yep

**Comment by 0xAaCE** [comment]
> @aleacevedo had a look, noticed the move to dfinity 0.0.10, and confirm the changes for the move of types from dfinity/candid (Blob to ArrayBuffer, etc), but haven't tested locally with a plug build.

Doing right now 🤞

File: src/utils/buffer.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+/**
+ * Concatenate multiple array buffers.
+ * @param buffers The buffers to concatenate.
+ */
+export function concat(...buffers: ArrayBuffer[]): ArrayBuffer {
```
Comment (jsonsivar) [inline src/utils/buffer.ts:5]: Is it okay that this takes in `ArrayBuffer` but on [using it](https://github.com/Psychedelic/plug-inpage-provider/pull/38/files#diff-e63da2fa283f44699b77e8122f4ed3d681617e86ad62105bee55bde60124ea78R88) we pass in a `Uint8Array`?

**Review by jsonsivar — APPROVED** [review]

**Pull request merged**